test/functionalTest/cases/3001_mqtt/subscribe_and_update.test

```diff
@@ -0,0 +1,129 @@
+# Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+MQTT Notifications
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 0-255 IPV4 -mqttHost localhost
+
+--SHELL--
+echo "01. POST /v2/entities, creating E1/T1"
+echo "====================================="
+payload='{
+  "id": "E1",
+  "type": "T1",
+  "A1": {
+    "value": 23,
+    "type": "Float"
+  },
+  "A2": {
+    "value": 720,
+    "type": "Integer"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+echo "02. Subscribe to E1 on condition A2"
+echo "==================================="
+payload='{
+  "description": "Test subscription",
+  "subject": {
+    "entities": [
+      {
+        "id": "E1",
+        "type": "T1"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "A2"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "mqtt://mqtttest.com"
```

Comment:
This doesn't follow the format described at https://github.com/telefonicaid/fiware-orion/issues/3001

```
"mqtt": {
  "topic": "/my/topic"
}
```

althought I guess is intentional, as the processing of that element is out of the scope of this PR (it is related with the `MqttInfo` discussion).

A bit off-topic, but... is `mqtt://` an standarized URL schema?

Reply:
Yes it is intentional as you commented. 

Regarding the schema, good point, thanks for bringing it up. The documentation on [mqtt.org](http://mqtt.org/documentation) points to [this page](https://github.com/mqtt/mqtt.github.io/wiki/URI-Scheme) which suggests the following format:

    mqtt[s]://[username][:password]@host.domain[:port]

I think we can use this format unless you see any problems with it.

Reply:
Not sure if that reference is yet a draft proposal or an official standard. Looking to IANA registry (https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml), it seems that "mqtt" is not there.

In addition, that schema seems to be aimed at defining the connection to the MQTT broker (which is solved in our case by `-mqttHost` and so on). It seems there isn't place for the topic in that URL.

In sum, I think the original idea of 

```
"mqtt": {
  "topic": "/my/topic"
}
```

is better (not for this PR, but for the next ones)

Reply:
Yes, it is merely a suggestion which is based on community discussions. 

Having said that, I am sorry for the confusion, I totally agree with the original idea and I had already planned and implemented the necessary changes to remove that line. 

I just kept it to have the module working for the first PR until we finalize the discussions about MqttInfo. Then I forget what I am doing, I guess I need a break :man_facepalming:

Reply:
Don't, worry, that's the point of having code review, detect this kind of things :D